interp/src/structures/names.rs

```diff
@@ -97,63 +94,84 @@ impl QualifiedInstanceName {
         string_vec.push(self.name.id.clone());
         string_vec.join(".").into()
     }
-}
 
-impl QualifiedInstanceName {
-    pub fn new(prefix: &ComponentQIN, name: &Id) -> Self {
+    pub fn new(prefix: &ComponentQualifiedInstanceName, name: &Id) -> Self {
         Self {
             prefix: prefix.clone(),
             name: name.clone(),
         }
     }
-}
 
-#[derive(Debug, Clone, Hash, Eq, PartialEq)]
-/// A qualified name which does not contain instance information
-pub struct QualifiedName {
-    prefix: Vec<Id>,
-    name: Id,
-}
+    pub fn prefix_length(&self) -> usize {
+        self.prefix.0.len()
+    }
 
-#[derive(Debug, Clone, Hash, Eq, PartialEq)]
-/// A qualified instance group name
-pub struct GroupQIN(QualifiedInstanceName);
+    pub fn shared_prefix_of(&self, other: &Self) -> bool {
+        for (a, b) in self.prefix.0.iter().zip(other.prefix.0.iter()) {
+            if a != b {
+                return false;
+            }
+        }
+        true
+    }
 
-impl GroupQIN {
-    pub fn new(prefix: &ComponentQIN, name: &Id) -> Self {
-        Self(QualifiedInstanceName::new(prefix, name))
+    pub fn equal_prefix(&self, other: &Self) -> bool {
+        self.prefix == other.prefix
     }
 }
 
-#[derive(Debug, Clone, Hash, Eq, PartialEq)]
-/// A qualified group name
-pub struct GroupQN(QualifiedName);
+#[derive(Debug)]
+pub enum GroupName {
+    /// An actual group
+    Group(Id),
+    /// A phantom group with a displayable name
+    Phantom(Id),
+    /// No group name
+    None,
```

Comment:
When does the none case arise?

Reply:
This shows up for components/subcomponents and is how they appear in the tree since any groups that are running in the given subcomponent should be a child of it. Also is how structural subcomponents appear.